readme.md

```diff
@@ -1,848 +1,1214 @@
 
-    ALWAYS evaluate for the best, most accurate result - the above is a guideline, not a dogma.
+```javascript
+// When evaluating that reference is null or undefined, prefer explicit comparison:
+if ( foo === null || foo === undefined ) ...
 
-    ```javascript
+// instead of == type coersion:
+if ( foo == null ) ...
+```
 
-    // 4.2.1
-    // Type coercion and evaluation notes
+#### 2.4.5. Loose equality coersion 
+Use loose equality operator to simplify your code when you want to leverage type coercion. Remember that strict equality comparator `===` is checking if both the type and the value you are comparing are the same. In a constract, loose equality operator `==` will try to do type coersion which may be useful in some cases when comparing values where types are less revelant than values.
 
-    // Use === only if you have to - for example when you're dealing
-    // with user input or you need to distinguish between null and undefined.
-    // You can safely use == in most cases, especially
-    // when function's input is documented.
-    // It will be caller's fault if he passes wrong arguments.
+```javascript
+false === 'false'
+// false
 
+false == 'false'
+// true
 
-    // 4.2.2
-    // Booleans, Truthies & Falsies
+// 1 === '1'
+// false
 
-    // Booleans:
-    true, false
+// 1 == '1'
+// true
+```
 
-    // Truthy:
-    "foo", 1
+#### 2.4.6. Booleans, Truthies & Falsies
+```javascript
+// Booleans:
+true, false
 
-    // Falsy:
-    "", 0, null, undefined, NaN, void 0
+// Truthy:
+"foo", 1
+
+// Falsy:
+"", 0, null, undefined, NaN, void 0
+```
+
+### 2.5. Practical Style
+
+#### 2.5.1. Module pattern
+
+```javascript
+( function() {
+	var Module = ( function() {
+		var data = "secret";
+
+		return {
+			// This is some boolean property.
+			bool: true,
+			// Some string value.
+			string: 'a string',
+			// An array property.
+			array: [ 1, 2, 3, 4 ],
+			// An object property.
+			object: {
+				lang: 'en-Us'
+			},
+
+			getData: function() {
+				// Get the current value of `data`.
+				return data;
+			},
+
+			setData: function( value ) {
+				this.value = value;
+			}
+		};
+	} )();
 
-    ```
+	// Other things might happen here.
+
+	// Expose our module to the global object.
+	CKEDITOR.module = Module;
+} )();
+```
+#### 2.5.2. Practical constructor
 
+```javascript
+( function() {
+	var fooBar = CKEDITOR.tools.createClass( {
+		$: function( foo ) {
+			this.foo = foo;
+		},
+		
+		proto: {
+			getFoo: function() {
+				return this.foo;
+			},
+
+			setFoo: function( val ) {
+				this.foo = val;
+			}
+		}
+	} );
+
+	// Expose our constructor to the global object.
+	CKEDITOR.fooBar = fooBar;
+} )();
+```
+
+#### 2.5.3. Prototype extension
+
+```javascript
+( function() {
+	var fooBar = CKEDITOR.tools.extend( FooBar.prototype, {
```

Comment:
```suggestion
	CKEDITOR.tools.extend( FooBar.prototype, {
```

When you extend prototype you probably not expect to assign it to any variable.
alternative example could start like:

```javascript
( function() {
	var FooBar = new CKEDITOR.tools.createClass( {
		// class definition
	} )
	
	CKEDITOR.tools.extend( FooBar.prototype, {
		getFoo: function() {
			return this.foo;
		},

		setFoo: function( val ) {
			this.foo = val;
		}
	} );

	// Expose our constructor with extended prototype to the global object.
	CKEDITOR.FooBar = FooBar;
} )();
```